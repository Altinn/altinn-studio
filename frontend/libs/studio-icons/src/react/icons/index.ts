export { AccordionIcon } from './AccordionIcon';
export { ArrayIcon } from './ArrayIcon';
export { BooleanIcon } from './BooleanIcon';
export { CheckboxIcon } from './CheckboxIcon';
export { CombinationIcon } from './CombinationIcon';
export { ConfirmationTaskIcon } from './ConfirmationTaskIcon';
export { DataTaskIcon } from './DataTaskIcon';
export { DefinitionIcon } from './DefinitionIcon';
export { ElementIcon } from './ElementIcon';
export { EndEventIcon } from './EndEventIcon';
export { FeedbackTaskIcon } from './FeedbackTaskIcon';
export { GiteaIcon } from './GiteaIcon';
export { GitHubIcon } from './GitHubIcon';
export { GroupIcon } from './GroupIcon';
export { LikertIcon } from './LikertIcon';
export { LongTextIcon } from './LongTextIcon';
export { NavBarIcon } from './NavBarIcon';
export { NumberIcon } from './NumberIcon';
export { ObjectIcon } from './ObjectIcon';
export { TextIcon } from './TextIcon';
export { PaymentDetailsIcon } from './PaymentDetailsIcon';
export { PaymentTaskIcon } from './PaymentTaskIcon';
export { PropertyIcon } from './PropertyIcon';
export { RadioButtonIcon } from './RadioButtonIcon';
export { ReferenceIcon } from './ReferenceIcon';
export { RepeatingGroupIcon } from './RepeatingGroupIcon';
export { SectionHeaderWarningIcon } from './SectionHeaderWarningIcon';
export { SelectIcon } from './SelectIcon';
export { ShortTextIcon } from './ShortTextIcon';
export { SignTaskIcon } from './SignTaskIcon';
export { SlackIcon } from './SlackIcon';
export { StringIcon } from './StringIcon';
export { TitleIcon } from './TitleIcon';
