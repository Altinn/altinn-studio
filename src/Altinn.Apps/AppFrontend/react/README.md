## Running the tests

### Lint checks

1. Navigate to the folder `src/Altinn.Apps/AppFrontend/react`.
2. Execute `npm ci`. This step is only nescessary if you have not already done it, or if you change branches.
3. Execute `npm run install-deps`. This step is only nescessary if you have not already done it, or if you change branches.
4. Execute `npm run lint`.

### Unit tests

1. Navigate to the folder `src/Altinn.Apps/AppFrontend/react`.
2. Execute `npm ci`. This step is only nescessary if you have not already done it, or if you change branches.
3. Execute `npm run install-deps`. This step is only nescessary if you have not already done it, or if you change branches.
4. Execute `npm run test`.
