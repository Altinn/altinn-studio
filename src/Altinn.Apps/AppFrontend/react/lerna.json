{
  "command": {
    "bootstrap": {
      "hoist": true
    }
  },
  "packages": [
    "./*"
  ],
  "version": "independent"
}
