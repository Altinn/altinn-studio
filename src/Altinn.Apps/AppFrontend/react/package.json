{
	"name": "app-frontend",
	"private": true,
	"scripts": {
		"install-deps": "lerna bootstrap --hoist --ci",
		"no-ci-hoist": "lerna bootstrap --hoist --no-ci",
		"install-deps-with-lock": "lerna bootstrap",
		"test": "lerna exec --concurrency 1 -- npm test",
		"install-all": "lerna exec -- npm ci",
		"clean-lock": "lerna exec -- rm ./package-lock.json",
		"clean-node-modules": "rm -rf ./node_modules && lerna exec --concurrency 1 -- rm -rf ./node_modules",
		"update-packages": "lerna exec --concurrency 1 -- npm update --dev",
		"list-outdated": "lerna run list-outdated"
	},
	"devDependencies": {
		"@typescript-eslint/eslint-plugin": "~4.8.2",
		"@typescript-eslint/parser": "~4.8.2",
		"eslint": "~7.14.0",
		"eslint-config-airbnb": "~18.2.1",
		"eslint-plugin-import": "~2.22.1",
		"eslint-plugin-jsx-a11y": "~6.4.1",
		"eslint-plugin-react": "~7.21.5",
		"eslint-plugin-react-hooks": "^4.2.0",
		"ini": "1.3.5",
		"lerna": "3.22.1"
	}
}
