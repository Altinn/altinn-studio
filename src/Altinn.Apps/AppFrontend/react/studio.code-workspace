{
	"folders": [
		{
			"path": "."
		}
  ],
  "settings": {
    "typescript.surveys.enabled": false,
    "typescript.tsdk": "node_modules/typescript/lib",
    "eslint.alwaysShowStatus": true,
    "eslint.validate": [
      "javascript",
      "javascriptreact",
      { "language": "typescript", "autoFix": true },
      { "language": "typescriptreact", "autoFix": true}
    ]
  }
}
