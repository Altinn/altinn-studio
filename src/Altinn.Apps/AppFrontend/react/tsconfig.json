{
  "compilerOptions": {
    "outDir": "compiled",
    "module": "commonjs",
    "target": "es5",
    "lib": [
      "es6",
      "dom",
      "es7",
      "esnext"
    ],
    "sourceMap": true,
    "allowJs": true,
    "jsx": "react",
    "moduleResolution": "node",
    "rootDir": "./",
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "strictNullChecks": false,
    "noUnusedLocals": true
  },
  "include": [
    "**/*.ts",
    "**/*.tsx",
  ],
  "exclude": [
    "node_modules",
    "build",
    "scripts",
    "acceptance-tests",
    "webpack",
    "jest",
    "setupTests.ts",
    "templates",
    "**/setupTests.ts"
  ],
}
