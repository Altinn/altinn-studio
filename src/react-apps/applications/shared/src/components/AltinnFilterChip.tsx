import { Chip, createMuiTheme } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import altinnTheme from '../theme/altinnStudioTheme';
import * as React from 'react';

export interface IAltinnFilterChipCompontentProvidedProps {
  classes: any;
  key: any;
  label: string;
  onclickFunction?: any;
  active: boolean;
  className?: string;
  deleteIcon?: any;
  onDeleteFunction?: any;
  sortIcon?: boolean;
}

export interface IAltinnFilterChipComponentState {
}

const theme = createMuiTheme(altinnTheme);

const styles = {
  chip: {
    color: '#000000',
    borderColor: theme.palette.primary.dark,
    borderWidth: '1px',
    borderStyle: 'dotted',
    backgroundColor: '#FFF',
    borderRadius: '36px',
    minHeight: '36px',
    fontSize: '16px',
    '&:hover': {
      backgroundColor: 'transparent !Important',
    },
    '&:focus': {
      backgroundColor: 'transparent !Important',
      borderColor: theme.palette.primary.main,
      borderWidth: '1px',
      borderStyle: 'solid',
    },
  },
  chipActive: {
    backgroundColor: theme.palette.primary.light + ' !Important',
    border: '1px solid ' + theme.palette.primary.dark,
    borderRadius: '36px',
    minHeight: '36px',
    fontSize: '16px',
    '&:hover': {
      backgroundColor: theme.palette.primary.light + ' !Important',
    },
    '&:focus': {
      backgroundColor: theme.palette.primary.light + ' !Important',
      border: '1px solid ' + theme.palette.primary.main,
    }
  },
  down: {
    borderTop: '10px solid black',
    borderLeft: '8px solid transparent',
    borderRight: '8px solid transparent',
    margin: '4px 8px 2px 0px',
  },
}

class AltinnFilterChip extends React.Component<IAltinnFilterChipCompontentProvidedProps, IAltinnFilterChipComponentState> {
  public render() {
    const { classes } = this.props;
    return (
      <Chip
        key={this.props.key}
        label={this.props.label}
        clickable={true}
        color='primary'
        variant='outlined'
        onClick={this.props.onclickFunction}
        onDelete={this.props.onDeleteFunction}
        deleteIcon={this.props.sortIcon ? <i className={classNames(classes.down)} /> : this.props.deleteIcon}
        className={classNames(
          this.props.className,
          classes.chip,
          { [classes.chipActive]: this.props.active })}
      />
    );
  }
}

export default withStyles(styles)(AltinnFilterChip);