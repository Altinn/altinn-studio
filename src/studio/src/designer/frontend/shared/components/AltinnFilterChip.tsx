import { Chip, createMuiTheme } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import * as React from 'react';
import altinnTheme from '../theme/altinnStudioTheme';

export interface IAltinnFilterChipComponentProvidedProps {
  classes: any;
  key: any;
  label: string;
  onclickFunction?: any;
  active: boolean;
  className?: string;
  deleteIcon?: any;
  onDeleteFunction?: any;
  sortIcon?: boolean;
}

export interface IAltinnFilterChipComponentState {
}

const theme = createMuiTheme(altinnTheme);

const styles = {
  chip: {
    'color': theme.altinnPalette.primary.black,
    'borderColor': theme.altinnPalette.primary.blueDark,
    'borderWidth': '1px',
    'borderStyle': 'dotted',
    'backgroundColor': theme.altinnPalette.primary.white,
    'borderRadius': '36px',
    'minHeight': '36px',
    'fontSize': '16px',
    '&:hover': {
      backgroundColor: 'transparent !Important',
    },
    '&:focus': {
      backgroundColor: 'transparent !Important',
      borderColor: theme.altinnPalette.primary.blueDark,
      borderWidth: '1px',
      borderStyle: 'solid',
    },
  },
  chipActive: {
    'backgroundColor': theme.altinnPalette.primary.blueLight + ' !Important',
    'border': '1px solid ' + theme.altinnPalette.primary.blueDark,
    'borderRadius': '36px',
    'minHeight': '36px',
    'fontSize': '16px',
    '&:hover': {
      backgroundColor: theme.altinnPalette.primary.blueLighter + ' !Important',
    },
    '&:focus': {
      backgroundColor: theme.altinnPalette.primary.blueLighter + ' !Important',
      border: '1px solid ' + theme.altinnPalette.primary.blueDark,
      boxShadow: '0px 0px 6px ' + theme.altinnPalette.primary.blueDark,
    },
  },
  down: {
    borderTop: '10px solid' + theme.altinnPalette.primary.black,
    borderLeft: '8px solid transparent',
    borderRight: '8px solid transparent',
    margin: '4px 8px 2px 0px',
  },
};

// eslint-disable-next-line max-len
export class AltinnFilterChip extends React.Component<IAltinnFilterChipComponentProvidedProps, IAltinnFilterChipComponentState> {
  public render() {
    const { classes } = this.props;
    return (
      <Chip
        key={this.props.key}
        label={this.props.label}
        clickable={true}
        color='primary'
        variant='outlined'
        onClick={this.props.onclickFunction}
        onDelete={this.props.onDeleteFunction}
        deleteIcon={this.props.sortIcon ? <i className={classNames(classes.down)} /> : this.props.deleteIcon}
        className={classNames(
          this.props.className,
          classes.chip,
          { [classes.chipActive]: this.props.active })}
      />
    );
  }
}

export default withStyles(styles)(AltinnFilterChip);
