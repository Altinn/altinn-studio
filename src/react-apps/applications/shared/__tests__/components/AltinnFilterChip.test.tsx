import 'jest';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import AltinnFilterChip from '../../src/components/AltinnFilterChip';

describe('>>> AltinnFilterChip--- Snapshot', () => {
  let mockKey: any;
  let mockLabel: string;
  let mockOnclickFunction: any;
  let mockActive: boolean;
  let mockClassName: string;
  let mockDeleteIcon: any;
  let mockOnDeleteFunction: any;
  let mockSortIcon: boolean;
  beforeEach(() => {
    mockKey = 0;
    mockLabel = 'mock label';
    mockActive = false;
    mockClassName = 'mock-class-name';
    mockSortIcon = true;
    mockDeleteIcon = <i className='ai ai-delete' />;
    mockOnDeleteFunction = () => {
      return false;
    };
    mockOnclickFunction = () => {
      return false;
    };
  });

  it('>>> Capture snapshot of AltinnFilterChip with the lest amount of params', () => {
    const rendered = renderer.create(
      <AltinnFilterChip
        key={mockKey}
        label={mockLabel}
        active={mockActive}
      />,
    );
    expect(rendered).toMatchSnapshot();
  });

  it('>>> Capture snapshot of AltinnFilterChip with sort icon visible', () => {
    const rendered = renderer.create(
      <AltinnFilterChip
        key={mockKey}
        label={mockLabel}
        active={mockActive}
        className={mockClassName}
        sortIcon={mockSortIcon}
        onclickFunction={mockOnclickFunction}
        onDeleteFunction={mockOnDeleteFunction}
      />,
    );
    expect(rendered).toMatchSnapshot();
  });

  it('>>> Capture snapshot of AltinnFilterChip with delete icon visible', () => {
    const rendered = renderer.create(
      <AltinnFilterChip
        key={mockKey}
        label={mockLabel}
        active={mockActive}
        onclickFunction={mockOnclickFunction}
        onDeleteFunction={mockOnDeleteFunction}
        deleteIcon={mockDeleteIcon}
      />,
    );
    expect(rendered).toMatchSnapshot();
  });
});
