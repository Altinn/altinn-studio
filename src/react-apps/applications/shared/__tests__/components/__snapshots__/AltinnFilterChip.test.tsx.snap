// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> AltinnFilterChip--- Snapshot +++ Should match snapshot with delete icon visible 1`] = `
<div
  className="MuiChip-root-4 MuiChip-colorPrimary-5 MuiChip-clickable-7 MuiChip-clickableColorPrimary-8 MuiChip-deletable-10 MuiChip-deletableColorPrimary-11 MuiChip-outlined-13 MuiChip-outlinedPrimary-14 AltinnFilterChip-chip-1"
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <span
    className="MuiChip-label-23"
  >
    mock label
  </span>
  <i
    className="ai ai-delete MuiChip-deleteIcon-24 MuiChip-deleteIconOutlinedColorPrimary-27"
    onClick={[Function]}
  />
</div>
`;

exports[`>>> AltinnFilterChip--- Snapshot +++ Should match snapshot with sort icon visible 1`] = `
<div
  className="MuiChip-root-4 MuiChip-colorPrimary-5 MuiChip-clickable-7 MuiChip-clickableColorPrimary-8 MuiChip-deletable-10 MuiChip-deletableColorPrimary-11 MuiChip-outlined-13 MuiChip-outlinedPrimary-14 mock-class-name AltinnFilterChip-chip-1"
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <span
    className="MuiChip-label-23"
  >
    mock label
  </span>
  <i
    className="AltinnFilterChip-down-3 MuiChip-deleteIcon-24 MuiChip-deleteIconOutlinedColorPrimary-27"
    onClick={[Function]}
  />
</div>
`;

exports[`>>> AltinnFilterChip--- Snapshot +++ Should match snapshot with the least amount of params 1`] = `
<div
  className="MuiChip-root-4 MuiChip-colorPrimary-5 MuiChip-clickable-7 MuiChip-clickableColorPrimary-8 MuiChip-outlined-13 MuiChip-outlinedPrimary-14 AltinnFilterChip-chip-1"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex={0}
>
  <span
    className="MuiChip-label-23"
  >
    mock label
  </span>
</div>
`;
